### PR TITLE
dataset(ncslgr): add initial loading script

### PR DESCRIPTION
**Pull request by AmitMY**
Its a small dataset, but its heavily annotated
https://www.bu.edu/asllrp/ncslgr.html

image

### CODE REVIEW

**Comment by AmitMY** [comment]
Tests fail for dataset `coarse_discourse`, not this dataset.
Can you please re-run the failed circle run?

File: datasets/ncslgr/ncslgr.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    """BuilderConfig for NCSLGR."""
+
+    videos: bool = True
+
+
+class NCSLGR(datasets.GeneratorBasedBuilder):
+    """NCSLGR: a small American Sign Language corpus annotated with non-manual features"""
+
+    VERSION = datasets.Version("0.7.0")
+
```
Comment (lhoestq) [inline datasets/ncslgr/ncslgr.py:58]:
```suggestion
    BUILDER_CONFIG_CLASS = NCSLGRConfig
```
We need to specify that there's a custom config here ;)

File: datasets/ncslgr/ncslgr.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+import re
+from dataclasses import dataclass
+
+from tqdm import tqdm
+
+import datasets
+
+
+_DESCRIPTION = """
+A small American Sign Language corpus annotated with non-manual features
```
Comment (lhoestq) [inline datasets/ncslgr/ncslgr.py:29]: Maybe you could add more details in the description ?

**Review by lhoestq — COMMENTED** [review]
Awesome :) 

I left a few comments

File: datasets/ncslgr/ncslgr.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            with open(eaf_file_path, "r", encoding="utf-8") as f:
+                content = f.read()
+
+                if self.config.videos:
+                    videos_relative = re.findall('RELATIVE_MEDIA_URL="(.*)"', content)
+                    videos = [os.path.join(videos_path, v[3:]) for v in videos_relative]
+
+                sentences = list(self._extract_sentences(content))
+
+            if (i % 7 == 0 and split == "test") or (i % 7 == 1 and split == "dev") or (i % 7 > 1 and split == "train"):
```
Comment (AmitMY) [inline datasets/ncslgr/ncslgr.py:150]: I'll also remove this split decision here

**Comment by yjernite** [comment]
Hi @AmitMY, sorry for leaving you hanging for a minute :) 

We've developed a new pipeline for adding datasets with a few extra steps, including adding a dataset card. You can find the full process [here](https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md)

Would you be up for adding the tags and description in the README.md so we can merge this cool dataset?

**Comment by AmitMY** [comment]
@lhoestq should be ready for another review :)

**Comment by lhoestq** [comment]
Awesome thank you !

It looks like the PR now includes changes from other PR that were previously merged. 
Feel free to create another branch and another PR so that we can have a clean diff.

**Comment by AmitMY** [comment]
Closing for #958

**Pull request closed without merging**